hoomd/md/pair/aniso.py

```diff
@@ -563,3 +572,425 @@ def type_shapes(self):
+
+        .. rubric:: Example:
+
+        .. code-block:: python
+
+            patchy.patches['A'] = [(1,0,0), (1,1,1)]
+
+"""
+
+
+class Patchy(AnisotropicPair):
+    r"""Pair potential made directional by an envelope.
+
+    Patchy potentials compute the  potential between particles with
+    angular patches with an implementation based on
+    `Beltran-Villegas et. al.`_. With distance between particles
+    :math:`r` and minimal angles between respective particle direction and
+    the displacement vector :math:`\theta_i`, :math:`\theta_j` the patchy potential
+    computes:
+
+    .. _Beltran-Villegas et. al.: https://dx.doi.org/10.1039/C3SM53136H
+
+    .. math::
+
+        U(r, \theta_i, \theta_j) = f(\theta_i, \alpha, \omega)
+        f(\theta_j, \alpha, \omega)
+        U_{\mathrm{pair}}(r)
+
+    where :math:`f` is an orientation-dependent factor of the patchy spherical
+    cap half-angle :math:`\alpha` and patch steepness :math:`\omega`.
+
+    .. math::
+        \begin{align}
+        f(\theta, \alpha, \omega) &= \frac{\big(1+e^{-\omega (\cos{\theta} -
+        \cos{\alpha}) }\big)^{-1} - f_{min}}{f_{max} - f_{min}}\\
+        f_{max} &= \big( 1 + e^{-\omega (1 - \cos{\alpha}) } \big)^{-1} \\
+        f_{min} &= \big( 1 + e^{-\omega (-1 - \cos{\alpha}) } \big)^{-1} \\
+        \end{align}
+
+    .. invisible-code-block: python
+        patchy = hoomd.md.pair.aniso.PatchyLJ(nlist = neighbor_list,
+                                              default_r_cut = 3.0)
+        pair_params = {'epsilon': 10, 'sigma': 1}
+
+        patchy.params.default = dict(pair_params = pair_params,
+                                     envelope_params = {'alpha': math.pi/4,
+                                                        'omega': 30})
+        patchy.patches.default = []
+        simulation.operations.integrator.forces = [patchy]
+
+    For multiple patches, every combination of patch and
+    We use PatchyLJ as the example.
+
+    To make specific patches
+
+    todo write example
+
+    .. py:attribute:: params
+
+        The Patchy potential parameters. A dictionary of dictionaries that
+        are passed on to the envelope and to the pair potential.
+
+        The dictionary has the following keys:
+
+        * ``envelope_params`` (`dict`, **required**)
+
+          * ``alpha`` (`float`) - patch half-angle :math:`\alpha` :math:`[\mathrm{rad}]`
+          * ``omega`` (`float`) - patch steepness :math:`\omega`
+
+
+        * ``pair_params`` (`dict`, **required**) -
+          passed to isotropic potential (see subclasses).
+
+        .. rubric:: Example:
+
+        .. code-block:: python
+
+            envelope_params = {'alpha': math.pi/4, 'omega': 30}
+            patchy.params[('A', 'A')] = dict(pair_params = pair_params,
+                                             envelope_params = envelope_params)
+
+        Type: `TypeParameter` [`tuple` [``particle_type``, ``particle_type``],
+        `dict`]
+
+    .. py:attribute:: patches
```

Comment:
This parameter name is too vague. More specific options, from my favorite to least favorite:
1. `directors`
2. `patch_directors`
3. `patch_directions`
4. `patch_vectors`
5. `patch_direction_vectors`.